toolkit/themes/shared/design-system/config.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* eslint-env node */
+
+const StyleDictionary = require("style-dictionary");
+const { fileHeader } = StyleDictionary.formatHelpers;
+
+/**
+ * CSS formatter that converts tokens that have a light and dark value
+ * to a light-dark() function. If we try to use the built-in CSS formatter,
+ * we will lose the "darkValue" or any other property that isn't "value".
+ */
+function lightDarkFormatter(args) {
+  let dictionary = args.dictionary;
+  let modifiedDictionary = dictionary.allTokens.map(token => {
+    // the `dictionary` object now has `usesReference()` and
+    // `getReferences()` methods. `usesReference()` will return true if
+    // the value has a reference in it. `getReferences()` will return
+    // an array of references to the whole tokens so that you can access their
+    // names or any other attributes.
+    if (token.original.dark) {
+      // Note: make sure to use `token.original.value` because
+      // `token.value` is already resolved at this point.
+      const lightRefs = dictionary.getReferences(token.original.value);
+      const darkRefs = dictionary.getReferences(token.original.dark);
+      if (lightRefs.length > 0 || darkRefs.length > 0) {
+        for (let i = 0; i < darkRefs.length; i++) {
+          let currentLightName = lightRefs[i]?.name;
+          let currentDarkName = darkRefs[i]?.name;
+          let lightValue = currentLightName ? 
+          `var(--${currentLightName})` : `${token.original.value}`;
+          let darkValue = currentDarkName ?
+          `var(--${currentDarkName})` : `${token.original.dark}`;
+          token.value = `light-dark(${lightValue}, ${darkValue})`
+        }
```

Comment:
```suggestion
          let currentLightName = lightRefs[0]?.name;
          let currentDarkName = darkRefs[0]?.name;
          let lightValue = currentLightName ? 
          `var(--${currentLightName})` : `${token.original.value}`;
          let darkValue = currentDarkName ?
          `var(--${currentDarkName})` : `${token.original.dark}`;
          token.value = `light-dark(${lightValue}, ${darkValue})`
```

Looking at [the source](https://github.com/amzn/style-dictionary/blob/17f4cb2f30bd002dfd55d6ef8c5bee4138de8d64/lib/utils/references/getReferences.js#L31) it seems like `getReferences` only returns an array for cases where a single value interpolates multiple variables, something like:

```json
"light": "light-dark({color-red-05}, {color-red-80})"
```
since that's not a thing for us at the moment we can probably avoid looping (plus using the length of `darkRefs` could hypothetically cause a bug if we had multiple interpolated variables in the light value but not the dark value)

Reply:
Works for me, thanks for looking into the source!